src/components/ListAllArticles.jsx

```diff
@@ -1,29 +1,53 @@
 import { useEffect, useState } from "react";
-import { useParams } from "react-router-dom";
+import { useParams, useSearchParams } from "react-router-dom";
 import * as API from "../api";
 import ArticleCard from "../components/ArticleCard";
+import SortAndOrderSection from "./SortAndOrderSection";
 
 const ListAllArticles = () => {
   const [isLoading, setIsLoading] = useState(true);
-  const [articleList, setArticleList] = useState("");
+  const [articleList, setArticleList] = useState([]);
+  const [selectSortValue, setSelectSortValue] = useState("created_at");
+  const [selectOrderValue, setSelectOrderValue] = useState("desc");
+  // const [searchParams, setSearchParams] = useSearchParams();
   const { topicName } = useParams();
 
   useEffect(() => {
     setIsLoading(true);
-
-    API.getAllArticles(topicName).then((articles) => {
-      setArticleList(articles);
-      setIsLoading(false);
-    });
-  }, [topicName]);
+    // setSearchParams({ sort_by: selectSortValue, order: selectOrderValue });
+    API.getAllArticles(topicName, selectSortValue, selectOrderValue).then(
+      (articles) => {
+        setArticleList(articles);
+        setIsLoading(false);
+      }
+    );
+  }, [topicName, selectSortValue, selectOrderValue]);
   return isLoading ? (
     <p>Loading ...</p>
   ) : (
     <section className="article-list">
       {topicName ? (
-        <h2> Articles in {`${topicName}`}</h2>
+        <>
+          <h2> Articles in {`${topicName}`}</h2>
+          <SortAndOrderSection
+            selectSortValue={selectSortValue}
+            setSelectSortValue={setSelectSortValue}
+            selectOrderValue={selectOrderValue}
+            setSelectOrderValue={setSelectOrderValue}
+          />
+        </>
       ) : (
-        <h2>All Articles</h2>
+        <>
+          {" "}
+          console.log(topic, sort_by, order, "in api");
```

Comment:
This line's coming up on the screen at the moment in the Articles tab, I'd get rid 🗑️